nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ControllerResource.java

```diff
@@ -2411,6 +2429,289 @@ public Date getEndDate() {
         }
     }
 
+    // ------------
+    // NARs
+    // ------------
+
+    @POST
+    @Consumes(MediaType.APPLICATION_OCTET_STREAM)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("nar-manager/upload")
```

Comment:
The reason I added `/upload` is because when we implement the second phase of this with the Extension Registry Clients, I picture there being another similar end-point, but instead of uploading it would be `/controller/nar-manager/install` where the body specifies a registry client id and coordinates to indicate to install the given NAR from the given client.

So we can do one of the following:

```
/nar-manager/upload
/nar-manager/install
```

```
/nar-manager/nars/upload
/nar-manager/nars/install
```

```
/nar-manager/nars  (this would be the upload without any specific path)
/nar-manager/nars/install
```
Thoughts?